### PR TITLE
update min python to 3.6 

### PR DESCRIPTION
**Pull request by edyoung**
This allows use of f-strings for interpolation
Updated one file (absyn.py) to see how that looks
Updated readme and test control files to specify new min

### CODE REVIEW

**Comment by edyoung** [comment]
Curious to see an apparently random segfault in the test runs. We must have an intermittent issue in there somewhere. Why does everything run on push and pull_request, is that useful? Guess it doesn;t hurt

**Comment by cjmayo** [comment]
> Curious to see an apparently random segfault in the test runs. We must have an intermittent issue in there somewhere. Why does everything run on push and pull_request, is that useful? Guess it doesn;t hurt

Actions give you a lot of control, which also means there is plenty to figure out! It appeared that you need the pull_request one to get a check when the PR is created, and the push both for updates to master and updates to a PR - but maybe updates to a PR can be done by the synchronise event (more investigation needed).

File: .github/workflows/test-gnofract4d.yml [hunk 1]
```diff
@@ -18,9 +18,6 @@ jobs:
           - os: ubuntu-18.04
             python: 3.6
             toxenv: py
-          - os: ubuntu-16.04
```
Comment (cjmayo) [inline .github/workflows/test-gnofract4d.yml:21]: Yes, 3.5 is the standard Python 3 for Ubuntu 16.04 so this does mean dropping it.

File: fract4d_compiler/absyn.py [hunk 2]
```diff
@@ -4,7 +4,6 @@
 
 from . import fracttypes
 
-
```
Comment (cjmayo) [inline fract4d_compiler/absyn.py:7]: Two lines here and before top-level functions are as per PEP 8:
https://www.python.org/dev/peps/pep-0008/#blank-lines

File: fract4d_compiler/absyn.py [hunk 3]
```diff
@@ -118,35 +105,30 @@ def __next__(self):
 
         return ret
 
-
```
Comment (cjmayo) [inline fract4d_compiler/absyn.py:121]: PEP 8 line as above.

File: fract4d_compiler/absyn.py [hunk 4]
```diff
@@ -118,35 +105,30 @@ def __next__(self):
...
         if not isinstance(tree.children, list):
-            raise Exception("children not a list: %s instead" % tree.children)
+            raise Exception(f"children not a list: {tree.children} instead")
         for child in tree.children:
             CheckTree(child, 0)
     return 1
 
 # shorthand named ctors for specific node types
 
-
```
Comment (cjmayo) [inline fract4d_compiler/absyn.py:136]: Run of PEP 8 lines as above.

File: tox.ini [hunk 5]
```diff
@@ -14,9 +15,10 @@ commands =
 
 passenv = DISPLAY XAUTHORITY HOME
 
+# imported by tests
```
Comment (cjmayo) [inline tox.ini:18]: It was specific to pytest so maybe reword to "pytest imported by tests"
